File: backend/infraction-service/app/com/radovan/play/modules/AutoBindModule.scala
package com.radovan.play.modules

import com.google.inject.AbstractModule
import com.radovan.play.brokers.{InfractionNatsListener, InfractionNatsSender}
import com.radovan.play.converter.TempConverter
import com.radovan.play.repository.InfractionRepository
import com.radovan.play.repository.impl.InfractionRepositoryImpl
import com.radovan.play.services.{EurekaRegistrationService, EurekaServiceDiscovery, InfractionService, PrometheusService}
import com.radovan.play.services.impl.{EurekaRegistrationServiceImpl, EurekaServiceDiscoveryImpl, InfractionServiceImpl, PrometheusServiceImpl}
import com.radovan.play.utils.{ServiceUrlProvider, TimeConversionUtils}
import io.micrometer.core.instrument.MeterRegistry
import io.micrometer.prometheusmetrics.{PrometheusConfig, PrometheusMeterRegistry}

class AutoBindModule extends AbstractModule{

  override def configure():Unit = {
    bind(classOf[EurekaServiceDiscovery]).to(classOf[EurekaServiceDiscoveryImpl]).asEagerSingleton()
    bind(classOf[EurekaRegistrationService]).to(classOf[EurekaRegistrationServiceImpl]).asEagerSingleton()
    bind(classOf[InfractionService]).to(classOf[InfractionServiceImpl]).asEagerSingleton()
    bind(classOf[PrometheusService]).to(classOf[PrometheusServiceImpl]).asEagerSingleton()

    bind(classOf[InfractionRepository]).to(classOf[InfractionRepositoryImpl]).asEagerSingleton()

    bind(classOf[InfractionNatsSender]).asEagerSingleton()
    bind(classOf[InfractionNatsListener]).asEagerSingleton()
    bind(classOf[TimeConversionUtils]).asEagerSingleton()
    bind(classOf[TempConverter]).asEagerSingleton()
    bind(classOf[ServiceUrlProvider]).asEagerSingleton()

    val prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT)
    bind(classOf[PrometheusMeterRegistry]).toInstance(prometheusRegistry)
    bind(classOf[MeterRegistry]).toInstance(prometheusRegistry)
  }

}

File: backend/registration-service/app/com/radovan/play/modules/AutoBindModule.java
package com.radovan.play.modules;

import com.google.inject.AbstractModule;
import com.radovan.play.brokers.NatsBrokerListener;
import com.radovan.play.converter.TempConverter;
import com.radovan.play.repositories.OwnerRepository;
import com.radovan.play.repositories.VehicleRepository;
import com.radovan.play.repositories.impl.OwnerRepositoryImpl;
import com.radovan.play.repositories.impl.VehicleRepositoryImpl;
import com.radovan.play.services.*;
import com.radovan.play.services.impl.*;
import com.radovan.play.utils.JwtUtil;
import com.radovan.play.utils.LoadData;
import com.radovan.play.utils.PublicKeyCache;
import com.radovan.play.utils.ServiceUrlProvider;
import io.micrometer.core.instrument.MeterRegistry;
import io.micrometer.prometheusmetrics.PrometheusConfig;
import io.micrometer.prometheusmetrics.PrometheusMeterRegistry;

public class AutoBindModule extends AbstractModule {

    @Override
    protected void configure(){
        bind(OwnerService.class).to(OwnerServiceImpl.class).asEagerSingleton();
        bind(VehicleService.class).to(VehicleServiceImpl.class).asEagerSingleton();
        bind(PrometheusService.class).to(PrometheusServiceImpl.class).asEagerSingleton();
        bind(EurekaServiceDiscovery.class).to(EurekaServiceDiscoveryImpl.class).asEagerSingleton();
        bind(EurekaRegistrationService.class).to(EurekaRegistrationServiceImpl.class).asEagerSingleton();


        bind(OwnerRepository.class).to(OwnerRepositoryImpl.class).asEagerSingleton();
        bind(VehicleRepository.class).to(VehicleRepositoryImpl.class).asEagerSingleton();


        bind(JwtUtil.class).asEagerSingleton();
        bind(PublicKeyCache.class).asEagerSingleton();
        bind(ServiceUrlProvider.class).asEagerSingleton();
        bind(LoadData.class).asEagerSingleton();
        bind(TempConverter.class).asEagerSingleton();
        bind(NatsBrokerListener.class).asEagerSingleton();
        PrometheusMeterRegistry prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
        bind(PrometheusMeterRegistry.class).toInstance(prometheusRegistry);
        bind(MeterRegistry.class).toInstance(prometheusRegistry);
    }
}
